adserver/views.py

```diff
@@ -751,6 +753,152 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class AdvertiserGeoReportView(AdvertiserAccessMixin, GeoReportMixin, BaseReportView):
+
+    """A report for an advertiser broken down by geo."""
+
+    model = GeoImpression
+    template_name = "adserver/reports/advertiser-geo.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        start_date = context["start_date"]
+        end_date = context["end_date"]
+
+        country = self.request.GET.get("country", "")
+
+        advertiser_slug = kwargs.get("advertiser_slug", "")
+        advertiser = get_object_or_404(Advertiser, slug=advertiser_slug)
+
+        queryset = self.get_queryset(
+            advertiser=advertiser,
+            start_date=start_date,
+            end_date=end_date,
+            country=country,
+        )
+
+        report = AdvertiserGeoReport(
+            queryset,
+            # Index by date if filtering report to a single country
+            index="date" if country else None,
+            order="-date" if country else None,
+            max_results=None if country else self.LIMIT,
+        )
+        report.generate()
+
+        country_options = self.get_country_options(
+            self.get_queryset(
+                advertiser=advertiser,
+                start_date=context["start_date"],
+                end_date=context["end_date"],
```

Comment:
This should probably use the `start_date` and `end_date` vars we already pulled out, or just access them directly in the context. I don't have a preference, just want it to be the same.